pkg/identity/identitymanager/manager_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package identitymanager
+
+import (
+	"testing"
+
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/labels"
+	. "gopkg.in/check.v1"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type IdentityManagerTestSuite struct{}
+
+var _ = Suite(&IdentityManagerTestSuite{})
+
+func (s *IdentityManagerTestSuite) TestIdentityManagerLifecycle(c *C) {
+
+	idFooSelectLabelArray := labels.ParseSelectLabelArray("id=foo")
+	idFooSelectLabels := labels.Labels{}
+	for _, lbl := range idFooSelectLabelArray {
+		idFooSelectLabels[lbl.Key] = lbl
+	}
+
+	idBarSelectLabelArray := labels.ParseSelectLabelArray("id=bar")
+	idBarSelectLabels := labels.Labels{}
+	for _, lbl := range idBarSelectLabelArray {
+		idBarSelectLabels[lbl.Key] = lbl
+	}
+
+	fooIdentity := identity.NewIdentity(identity.NumericIdentity(12345), idFooSelectLabels)
+	barIdentity := identity.NewIdentity(identity.NumericIdentity(54321), idBarSelectLabels)
+
+	idm := NewIdentityManager()
+	c.Assert(idm.identities, Not(IsNil))
+
+	_, exists := idm.identities[fooIdentity.ID]
+	c.Assert(exists, Equals, false)
+
+	idm.Add(fooIdentity)
+	c.Assert(idm.identities[fooIdentity.ID].refCount, Equals, uint(1))
```

Comment:
I was going to suggest creating a `GetRefcount(id Identity) uint` function and test with that, but it is basically this line of code. One thing I'm not sure about is that the only way we export the refcount is via `GetIdentityModels` but we aren't testing that here (and it would be a pain to assert on its members).